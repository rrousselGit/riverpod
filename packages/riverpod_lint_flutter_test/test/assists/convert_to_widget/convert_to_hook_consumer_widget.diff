Message: `Convert to HookConsumerWidget`
Priority: 37
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:5`:
```
import 'package:flutter_hooks/flutter_hooks.dart';

- class Stateless extends StatelessWidget {
-   const Stateless({super.key});
- 
-   @override
-   Widget build(BuildContext context) {
+ class Stateless extends HookConsumerWidget {
+   const Stateless({super.key});
+ 
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
```
---
Message: `Convert to HookConsumerWidget`
Priority: 37
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:14`:
```
}

- class StatelessWithComma extends StatelessWidget {
-   const StatelessWithComma({super.key});
- 
-   @override
-   Widget build(
-     BuildContext context,
+ class StatelessWithComma extends HookConsumerWidget {
+   const StatelessWithComma({super.key});
+ 
+   @override
+   Widget build(
+     BuildContext context, WidgetRef ref,
  ) {
    return const Placeholder();
```
---
Message: `Convert to HookConsumerWidget`
Priority: 37
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:25`:
```
}

- class Hook extends HookWidget {
-   const Hook({super.key});
- 
-   @override
-   Widget build(BuildContext context) {
+ class Hook extends HookConsumerWidget {
+   const Hook({super.key});
+ 
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
```
---
Message: `Convert to HookConsumerWidget`
Priority: 33
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:46`:
```
}

- class Stateful extends StatefulWidget {
-   const Stateful({super.key});
- 
-   @override
-   State<Stateful> createState() => _StatefulState();
- }
- 
- class _StatefulState extends State<Stateful> {
-   /// Hello world
-   @override
-   Widget build(BuildContext context) {
+ class Stateful extends HookConsumerWidget {
+   const Stateful({super.key});
+ 
+  
+   /// Hello world
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    // Some comments
    return const Placeholder();
```
---
Message: `Convert to HookConsumerWidget`
Priority: 33
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:62`:
```
}

- class ExplicitCreateState extends StatefulWidget {
-   const ExplicitCreateState({super.key});
- 
-   @override
-   ExplicitCreateStateState createState() => ExplicitCreateStateState();
- }
- 
- class ExplicitCreateStateState extends State<ExplicitCreateState> {
-   @override
-   Widget build(
-     BuildContext context,
-   ) {
+ class ExplicitCreateState extends HookConsumerWidget {
+   const ExplicitCreateState({super.key});
+ 
+  
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
```
---
Message: `Convert to HookConsumerWidget`
Priority: 37
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:78`:
```
}

- class HookStateful extends StatefulHookWidget {
-   const HookStateful({super.key});
- 
-   @override
-   State<HookStateful> createState() => HookStatefulState();
- }
- 
- class HookStatefulState extends State<HookStateful> {
-   @override
-   Widget build(BuildContext context) {
+ class HookStateful extends HookConsumerWidget {
+   const HookStateful({super.key});
+ 
+  
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
```
---
Message: `Convert to HookConsumerWidget`
Priority: 37
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:92`:
```
}

- class ConsumerStateful extends ConsumerStatefulWidget {
-   const ConsumerStateful({super.key});
- 
-   @override
-   ConsumerState<ConsumerStateful> createState() => _ConsumerStatefulState();
- }
- 
- class _ConsumerStatefulState extends ConsumerState<ConsumerStateful> {
-   @override
-   Widget build(BuildContext context) {
+ class ConsumerStateful extends HookConsumerWidget {
+   const ConsumerStateful({super.key});
+ 
+  
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
```
---
Message: `Convert to HookConsumerWidget`
Priority: 37
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:106`:
```
}

- class HookConsumerStateful extends StatefulHookConsumerWidget {
-   const HookConsumerStateful({super.key});
- 
-   @override
-   ConsumerState<HookConsumerStateful> createState() =>
-       _HookConsumerStatefulState();
- }
- 
- class _HookConsumerStatefulState extends ConsumerState<HookConsumerStateful> {
-   @override
-   Widget build(BuildContext context) {
+ class HookConsumerStateful extends HookConsumerWidget {
+   const HookConsumerStateful({super.key});
+ 
+  
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
```
---
Message: `Convert to HookConsumerWidget`
Priority: 37
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:121`:
```
}

- class Consumer extends ConsumerWidget {
+ class Consumer extends HookConsumerWidget {
  const Consumer({super.key});

```
---
Message: `Convert to HookConsumerWidget`
Priority: 37
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:133`:
```
}

- class StatelessWithField extends StatelessWidget {
-   const StatelessWithField({
-     super.key,
-     required this.field,
-   });
- 
-   final int field;
-   static final int staticField = 42;
- 
-   @override
-   Widget build(BuildContext context) {
+ class StatelessWithField extends HookConsumerWidget {
+   const StatelessWithField({
+     super.key,
+     required this.field,
+   });
+ 
+   final int field;
+   static final int staticField = 42;
+ 
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    return Column(
      children: [
```
---
Message: `Convert to HookConsumerWidget`
Priority: 37
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:180`:
```
}

- class ConsumerStatefulWithField extends ConsumerStatefulWidget {
-   const ConsumerStatefulWithField({
-     super.key,
-     required this.field,
-     required this.foo,
-   });
- 
-   final int field;
-   final FooClass foo;
-   static final int staticField = _constantNumber;
- 
-   @override
-   ConsumerState<ConsumerStatefulWithField> createState() =>
-       _ConsumerStatefulWithFieldState();
- }
- 
- const _constantNumber = 42;
- 
- class _ConsumerStatefulWithFieldState
-     extends ConsumerState<ConsumerStatefulWithField> {
-   void printFoo() {
-     print(widget.foo);
-   }
- 
-   @override
-   Widget build(BuildContext context) {
-     print(widget.field);
-     printFoo();
-     return Column(
-       children: [
-         Text('${widget.field}'),
-         Text('${widget.foo.bar}'),
-         Text('${ConsumerStatefulWithField.staticField}'),
-       ],
-     );
-   }
- }
+ class ConsumerStatefulWithField extends HookConsumerWidget {
+   const ConsumerStatefulWithField({
+     super.key,
+     required this.field,
+     required this.foo,
+   });
+ 
+   final int field;
+   final FooClass foo;
+   static final int staticField = _constantNumber;
+ 
+  
+   void printFoo() {
+     print(foo);
+   }
+ 
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
+     print(field);
+     printFoo();
+     return Column(
+       children: [
+         Text('$field'),
+         Text('${foo.bar}'),
+         Text('$staticField'),
+       ],
+     );
+   }
+ }
+ 
+ const _constantNumber = 42;

```
---
