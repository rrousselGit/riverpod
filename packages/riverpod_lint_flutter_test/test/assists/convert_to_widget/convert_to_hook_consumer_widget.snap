Message: `Convert to HookConsumerWidget`
<<<< start: 168 -- end: 265
StatelessWidget {
  const Stateless({super.key});

  @override
  Widget build(BuildContext contex
==== start: 168 -- end: 283
HookConsumerWidget {
  const Stateless({super.key});

  @override
  Widget build(BuildContext context, WidgetRef re
>>>>
Message: `Convert to HookConsumerWidget`
<<<< start: 342 -- end: 453
StatelessWidget {
  const StatelessWithComma({super.key});

  @override
  Widget build(
    BuildContext contex
==== start: 342 -- end: 471
HookConsumerWidget {
  const StatelessWithComma({super.key});

  @override
  Widget build(
    BuildContext context, WidgetRef re
>>>>
Message: `Convert to HookConsumerWidget`
<<<< start: 524 -- end: 607
Widget {
  const Hook({super.key});

  @override
  Widget build(BuildContext contex
==== start: 524 -- end: 630
ConsumerWidget {
  const Hook({super.key});

  @override
  Widget build(BuildContext context, WidgetRef re
>>>>
Message: `Convert to HookConsumerWidget`
<<<< start: 876 -- end: 1166
StatefulWidget {
  const Stateful({super.key});

  @override
  State<Stateful> createState() => _StatefulState();
}

class _StatefulState extends State<Stateful> {
  /// Hello world
  @override
  Widget build(BuildContext context) {
    // Some comments
    return const Placeholder();
  }

==== start: 876 -- end: 1073
HookConsumerWidget {
  const Stateful({super.key});

  
/// Hello world
  @override
  Widget build(BuildContext context, WidgetRef ref) {
    // Some comments
    return const Placeholder();
  }
}

>>>>
Message: `Convert to HookConsumerWidget`
<<<< start: 1203 -- end: 1514
StatefulWidget {
  const ExplicitCreateState({super.key});

  @override
  ExplicitCreateStateState createState() => ExplicitCreateStateState();
}

class ExplicitCreateStateState extends State<ExplicitCreateState> {
  @override
  Widget build(
    BuildContext context,
  ) {
    return const Placeholder();
  }

==== start: 1203 -- end: 1381
HookConsumerWidget {
  const ExplicitCreateState({super.key});

  
@override
  Widget build(
    BuildContext context, WidgetRef ref,
  ) {
    return const Placeholder();
  }
}

>>>>
Message: `Convert to HookConsumerWidget`
<<<< start: 1544 -- end: 1817
StatefulHookWidget {
  const HookStateful({super.key});

  @override
  State<HookStateful> createState() => HookStatefulState();
}

class HookStatefulState extends State<HookStateful> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }

==== start: 1544 -- end: 1706
HookConsumerWidget {
  const HookStateful({super.key});

  
@override
  Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
}

>>>>
Message: `Convert to HookConsumerWidget`
<<<< start: 1851 -- end: 2166
ConsumerStatefulWidget {
  const ConsumerStateful({super.key});

  @override
  ConsumerState<ConsumerStateful> createState() => _ConsumerStatefulState();
}

class _ConsumerStatefulState extends ConsumerState<ConsumerStateful> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }

==== start: 1851 -- end: 2017
HookConsumerWidget {
  const ConsumerStateful({super.key});

  
@override
  Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
}

>>>>
Message: `Convert to HookConsumerWidget`
<<<< start: 2204 -- end: 2549
StatefulHookConsumerWidget {
  const HookConsumerStateful({super.key});

  @override
  ConsumerState<HookConsumerStateful> createState() =>
      _HookConsumerStatefulState();
}

class _HookConsumerStatefulState extends ConsumerState<HookConsumerStateful> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }

==== start: 2204 -- end: 2374
HookConsumerWidget {
  const HookConsumerStateful({super.key});

  
@override
  Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
}

>>>>
Message: `Convert to HookConsumerWidget`
<<<< start: 2575 -- end: 2575
==== start: 2575 -- end: 2579
Hook
>>>>
